#!/usr/bin/env python

PACKAGE = "gr_thermal_processing"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("dilate_factor", int_t, 0 , "Dilate Factor",1, 0,20)
gen.add("erosion_factor", int_t, 0 , "Erosion Factor",1, 0,40)
gen.add("anchor_point", int_t, 0 , "Anchor Point x=y",-1, -1,10)
gen.add("ddepth", int_t, 0 , "Desired Depth",-1, -1,5)
gen.add("delta", int_t, 0 , "Noise after filter",0, 0,100)
gen.add("kernel_size", int_t, 0 , "Kernel Size",3, 3,13)
gen.add("filter_iterations", int_t, 0 , "Number of iterations",0, 0,30)
gen.add("threshold", int_t, 0 , "threshold",10, 0,255)
gen.add("threshold_mode", int_t, 0 , "threshold mode",3, 0,4)
gen.add("apply_threshold",  bool_t,   0, "Apply threshold",  True)
gen.add("norm_factor", double_t, 0 , "Voxel Cube Size", 0.5, 0,50.0)
exit(gen.generate(PACKAGE, "gr_thermal_processing", "Thermal"))
